cpu/cortexm_common/thread_arch.c

```diff
@@ -395,6 +397,7 @@ void __attribute__((naked)) __attribute__((used)) isr_pendsv(void) {
     "ldr    r0, [r0]                  \n" /* dereference TCB */
     "ldr    r0, [r0]                  \n" /* load tcb-sp to R0 */
     "mov    sp, r0                    \n" /* make user mode SP active SP */
+    "ibs                              \n" /* flush pipeline */
```

Comment:
I think this is only necessary when using msr, not by writing to sp in general. (gcc generates tons of sp modifying code in the function prologues, I've not seen a pipeline flush in there).